From c19ea8ab803612fbcf43338f8b27ea1f1e1fa1aa Mon Sep 17 00:00:00 2001
From: Jarkko Nikula <jarkko.nikula@linux.intel.com>
Date: Fri, 9 Jun 2017 14:45:43 +0300
Subject: [PATCH 0847/2302] Revert "mfd: intel-lpss: Avoid resuming
 runtime-suspended lpss unnecessarily"

This reverts commit d4688de00e18f1286122957e5f868c3091156bc3.

It is causing warnings from clk_core_disable() and clk_core_unprepare()
in drivers/i2c/busses/i2c-designware-platdrv.c due to a possibility to
enter suspend callback twice.

We don't have a clear or accepted fix for that. Best candidate so far is
unfortunately causing regression to previous platforms so it's better to
hold the patch before those are reworked.

Fixes: LCK-3500
Signed-off-by: Jarkko Nikula <jarkko.nikula@linux.intel.com>
Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
---
 drivers/mfd/intel-lpss.c | 9 ---------
 include/linux/pm.h       | 7 -------
 2 files changed, 16 deletions(-)

diff --git a/drivers/mfd/intel-lpss.c b/drivers/mfd/intel-lpss.c
index 52e6962..40e8d9b 100644
--- a/drivers/mfd/intel-lpss.c
+++ b/drivers/mfd/intel-lpss.c
@@ -485,15 +485,6 @@ static int resume_lpss_device(struct device *dev, void *data)
 int intel_lpss_prepare(struct device *dev)
 {
 	/*
-	 * This is safe because:
-	 * 1. The runtime suspend and system suspend
-	 * are of the same hook.
-	 * 2. This device is neither runtime wakeup source
-	 * nor system wakeup source.
-	 */
-	if (pm_runtime_status_suspended(dev))
-		return DPM_DIRECT_COMPLETE;
-	/*
 	 * Resume both child devices before entering system sleep. This
 	 * ensures that they are in proper state before they get suspended.
 	 */
diff --git a/include/linux/pm.h b/include/linux/pm.h
index cdfc82af..a0ceecc 100644
--- a/include/linux/pm.h
+++ b/include/linux/pm.h
@@ -788,11 +788,4 @@ enum dpm_order {
 	DPM_ORDER_DEV_LAST,
 };
 
-/*
- * Return this from system suspend/hibernation ->prepare() callback to
- * request the core to leave the device runtime-suspended during system
- * suspend if possible.
- */
-#define DPM_DIRECT_COMPLETE 1
-
 #endif /* _LINUX_PM_H */
-- 
2.7.4

